crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -63,26 +63,42 @@ const.GLOBAL_INPUTS_SECTION_OFFSET=400
 # The memory address at which the transaction reference block's commitment is stored.
 const.BLOCK_COMMITMENT_PTR=400
 
-# The memory address at which the account ID felts are stored.
-const.ACCT_ID_PTR=404
+# The memory address at which the native account ID is stored.
+const.NATIVE_ACCT_ID_PTR=404
 
 # The memory address at which the initial account commitment is stored.
 const.INIT_ACCOUNT_COMMITMENT_PTR=408
 
-# The memory address at which the input notes commitment is stored.
-const.INPUT_NOTES_COMMITMENT_PTR=412
-
 # The memory address at which the initial nonce is stored.
-const.INIT_NONCE_PTR=416
+const.INIT_NONCE_PTR=412
+
+# The memory address at which the initial vault root of the native account is stored.
+const.INIT_NATIVE_ACCOUNT_VAULT_ROOT_PTR=416
+
+# The memory address at which the initial storage commitment of the native account is stored.
+const.INIT_NATIVE_ACCOUNT_STORAGE_COMMITMENT_PTR=420
+
+# The memory address at which the initial code commitment of the native account is stored.
+# 
+# It points to the code commitment stored in the native account data block because the account's 
+# code cannot be changed during the user code execution, only in the epilogue, so the initial code 
+# commitment will be equal to the native account code commitment whenever it is called by a user.
+#
+# The value of this constant is equal to NATIVE_ACCOUNT_DATA_PTR + ACCT_CODE_COMMITMENT_OFFSET, but
+# because this constant should be created first, we have to use the final value itself.
+const.INIT_NATIVE_ACCOUNT_CODE_COMMITMENT_PTR=8204
```

Comment:
Does the order of declarations still matter? I thought with some of the refactoring in the assembler we could use constants out of order.